src/suspendplugin.c

```diff
@@ -238,6 +238,63 @@ suspend_set_wowlan(
     return err;
 }
 
+/**
+ * @brief Set powersave state (on/off), same as calling `iw dev %ifname% set power_save %is_enable%`
+ * 
+ * @param ifname interface name (e.g. wlan0)
+ * @param is_enable enable/disable state
+ * @return int 0 on success / errcode otherwise
+ */
+static
+int
+suspend_set_powersave(
+    const char *ifname, gboolean is_enable)
+{
+    int err = 0;
+    struct nl_msg *msg;
+
+    enum nl80211_ps_state ps_state;
+
+    int ifindex = 0;
+
+    ifindex = if_nametoindex(ifname);
+
+    if (ifindex == 0) {
```

Comment:
This same sequence is used in many places, should we turn it into a macro or function?

Reply:
I'm really not sure about that. To be honest I think that we should refactor that plugin as soon as we will have more devices which we are supporting using that plugin. Lets keep it as is for now.

Reply:
@monich @krnlyng  Can we keep it as is? 

I'm really not sure that part of code listed above should be moved to function or/macro.